packages/core/src/lib/filter/index.ts

```diff
@@ -335,12 +125,13 @@ class Filter extends BaseProtocol implements IReceiver {
    * Instead, use `createSubscription` to create a new subscription.
    */
   async subscribe<T extends IDecodedMessage>(
-    decoders: IDecoder<T> | IDecoder<T>[],
-    callback: Callback<T>
+    decoders: IDecoder<T> | IDecoder<T>[]
   ): Promise<Unsubscribe> {
-    const subscription = await this.createSubscription();
+    const decodersArray = Array.isArray(decoders) ? decoders : [decoders];
+
+    const subscription = await this.createSubscription(decodersArray);
 
-    await subscription.subscribe(decoders, callback);
+    await subscription.subscribe(decoders);
```

Comment:
Since it is done in `subscription` it is not needed here.

Reply:
I think you misunderstood my comment:

in `filter.subscribe`
```ts
    const subscription = await this.createSubscription(decodersArray);
    await subscription.subscribe(decoders);
```

`this.createSubscription(decodersArray)` is doing `subscription.subscribe(decoders)` under the hood